buildlog/buildlog_test.go

```diff
@@ -0,0 +1,201 @@
+package buildlog
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type writeCloser struct {
+	io.Writer
+}
+
+func (wc writeCloser) Close() error {
+	return nil
+}
+
+func write(tb testing.TB, filepath string, inputs []string) {
+	tb.Helper()
+
+	for _, input := range inputs {
+		file, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		_, err = file.Write([]byte(input))
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		err = file.Close()
+		if err != nil {
+			tb.Fatal(err)
+		}
+	}
+}
+
+func TestRun(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		tmpFile, err := ioutil.TempFile("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer tmpFile.Close()
+
+		testInputs := []string{
```

Comment:
We can make this `testInputs` to common.

Reply:
I fixed it.